src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -50,7 +50,17 @@ static void HandleNodeResolve(void * context, DnssdService * result, const Span<
     }
 
     DiscoveredNodeData nodeData;
-    result->ToDiscoveredNodeData(addresses, nodeData);
+
+    if (0 == strncmp(result->mType, kOperationalServiceName, sizeof(kOperationalServiceName)))
+    {
+        result->ToDiscoveredOperationalNodeBrowseData(nodeData);
+
+        nodeData.Get<OperationalNodeBrowseData>().LogDetail();
+        discoveryContext->OnNodeDiscovered(nodeData);
```

Comment:
You have to call `discoveryContext->Release()` before returning.

Reply:
That would release the context, and the following browse results will not have context or delegate. I think releasing makes sense for one shot query, does it makes sense to release for continuous query? 
When Stop is called the context gets released anyways right?

Reply:
This is reference counted object and `Release()` decrements the reference count which was increased for each service before this function is called (otherwise, there will be a context leak). Until `HandleNodeBrowse` is called with `finalBrowse=true`, the object is not actually released.

Reply:
@Damian-Nordic is correct: all codepaths through here need to call Release().

Reply:
fixed it.